lambda/shared/src/main/scala-3/feral/lambda/TracedHandlerPlatform.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2021 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package feral.lambda
+
+import cats.effect.{Trace as _, *}
+import cats.mtl.Local
+import natchez.*
+
+trait TracedHandlerPlatform:
+  def apply[F[_]: MonadCancelThrow, Event, Result](entryPoint: EntryPoint[F])(
+      handler: Trace[F] ?=> F[Option[Result]])(
```

Comment:
Splitting the `apply` method into Scala version-specific files lets us use a context function here, which improves the experience on Scala 3.